File: DataStructures/Tree/HuffmanDecode.py
import queue as Queue

"""class Node:
    def __init__(self, freq,data):
        self.freq= freq
        self.data=data
        self.left = None
        self.right = None
"""        

# Enter your code here. Read input from STDIN. Print output to STDOUT
def decodeHuff(root, s):
    p = root
    for i in s:
        if i == "0":
            p = p.left
            if p.left is None and p.right is None:
                print(p.data, end = "")
                p = root
        else:
            p = p.right
            if p.left is None and p.right is None:
                print(p.data, end = "")
                p = root

File: Data Structures Problems(C++)/Tree/HuffmanDecode.cpp
//
//  main.cpp
//  Huffman
//
//  Created by Vatsal Chanana

#include <stddef.h>
#include <iostream>
#include<queue>

using namespace std;

typedef struct node {
    int freq;
    char data;
    node * left;
    node * right;
} node;




void decode_huff(node * root,string s)
{
    node* temp = root;
    for (char c : s) {
        temp = c == '0' ? temp->left : temp->right;
        if (temp->data) {
            cout << temp->data;
            temp = root;
        }
    }
}

int main() {